addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -122,250 +130,75 @@ def reset_efaktur(self):
-
-            # Values now being corrected, we can format them for the CSV
-            for sale in sales:
-                sale.update({
-                    'HARGA_TOTAL': float_repr(sale['HARGA_TOTAL'], idr.decimal_places),
-                    'DPP': float_repr(sale['DPP'], idr.decimal_places),
-                    'PPN': float_repr(sale['PPN'], idr.decimal_places),
-                })
-
-            output_head += _csv_row(fk_values_list, delimiter)
-            for sale in sales:
-                of_values_list = ['OF'] + [str(sale[f]) for f in OF_HEAD_LIST[1:-2]] + ['0', '0']
-                output_head += _csv_row(of_values_list, delimiter)
-
-        return output_head
-
     def _prepare_etax(self):
         # These values are never set
         return {'JUMLAH_PPNBM': 0, 'UANG_MUKA_PPNBM': 0, 'JUMLAH_BARANG': 0, 'TARIF_PPNBM': 0, 'PPNBM': 0}
 
-    def _generate_efaktur(self, delimiter):
-        if self.filtered(lambda x: not x.l10n_id_kode_transaksi):
-            raise UserError(_('Some documents don\'t have a transaction code'))
-        if self.filtered(lambda x: x.move_type != 'out_invoice'):
-            raise UserError(_('Some documents are not Customer Invoices'))
-
-        output_head = self._generate_efaktur_invoice(delimiter)
-        my_utf8 = output_head.encode("utf-8")
-        out = base64.b64encode(my_utf8)
-
-        attachment = self.env['ir.attachment'].create({
-            'datas': out,
-            'name': 'efaktur_%s.csv' % (fields.Datetime.to_string(fields.Datetime.now()).replace(" ", "_")),
-            'type': 'binary',
-        })
-
-        for record in self:
-            record.message_post(attachment_ids=[attachment.id])
-        self.l10n_id_attachment_id = attachment.id
-        return {
-            'type': 'ir.actions.client',
-            'tag': 'reload',
-        }
+    def button_draft(self):
+        # EXTENDS 'account'
+        # When resetting to draft, we want the invoice to be removed from the document they are linked to.
+        # That document will be regenerated when downloading later on with only the remaining invoices.
+        invoices_with_document = self.filtered(lambda i: i.l10n_id_efaktur_document)
+        if invoices_with_document:
+            invoices_document = invoices_with_document.l10n_id_efaktur_document
+
+            invoices_document.attachment_id.unlink()
+            invoices_with_document.l10n_id_efaktur_document = False
+
+            empty_documents = invoices_document.filtered(lambda d: not d.invoice_ids)
+            # We would like to keep them in case the documents in the chatter are important.
+            # Users can always delete them manually as needed.
+            if empty_documents:
+                empty_documents.active = False
+
+            body = _("This invoice has been unlinked from the e-faktur document %s following the reset to draft.", invoices_document._get_html_link(title=f"{invoices_document.id}"))
```

Comment:
if possible it's always better to name the params

```suggestion
            body = _("This invoice has been unlinked from the e-faktur document %(document_link)s following the reset to draft.", 
                     document_link=invoices_document._get_html_link(title=f"{invoices_document.id}"))
```